src/modules/ekf2/ekf2_main.cpp

```diff
@@ -858,11 +858,14 @@ void Ekf2::run()
 			ev_data.quat = q;
 
 			// position measurement error from parameters. TODO : use covariances from topic
-			ev_data.posErr = _ev_pos_noise.get();
+			ev_data.posErr = fmaxf(_ev_pos_noise.get(), fmaxf(ev_pos.eph, ev_pos.epv));
 			ev_data.angErr = _ev_ang_noise.get();
 
-			// use timestamp from external computer, clocks are synchronized when using MAVROS
-			_ekf.setExtVisionData(vision_position_updated ? ev_pos.timestamp : ev_att.timestamp, &ev_data);
+			// only set data if all positions and velocities are valid
+			if (ev_pos.xy_valid && ev_pos.z_valid && ev_pos.v_xy_valid && ev_pos.v_z_valid) {
```

Comment:
I've checked the ecl EKF code and the result of stopping EV messages should be that after 5 seconds a "EKF External Vision Data Stopped" information message is sent to the console. If EV data is the only aiding source, then the EKF should fall back to a attitude and vertical pos/vel estimation only mode together with a "EKF stopping navigation" warning message.

This needs to be verified with testing before merging these changes.

Reply:
I'll test it in simulation and will share the logs. Thanks!